Automattic/wp-calypso
# Jetpack Agency Dashboard: Use jetpack_partner_types to detect agencies

#### Proposed Changes

A new `jetpack_partner_types` array was added to the response that comes from the `/me`. This PR updates the dashboard to use this boolean to determine if the user is an agency, and also performs this check in the routing part instead of the rendering part to improve initial load time.

#### Testing Instructions



1. Apply D83119-code to your sandbox and point public-api.wordpress.com to your sandbox.
2. Load `jetpack.cloud.localhost:3001/dashboard` and verify that the dashboard loads and renders appropriately.
3. With an agency user, visit `jetpack.cloud.localhost:3001/` and verify that you are redirected to `jetpack.cloud.localhost:3001/dashboard`.
4. With a non-agency user, visit `jetpack.cloud.localhost:3001/dashboard` and verify that you are redirected to ``jetpack.cloud.localhost:3001/`
 


Related to 1202076982646589-as-1202427818023261
